src/org/omegat/filters2/moodlephp/MoodlePHPFilter.java

```diff
@@ -0,0 +1,218 @@
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.omegat.filters2.AbstractFilter;
+import org.omegat.filters2.FilterContext;
+import org.omegat.filters2.Instance;
+import org.omegat.filters2.TranslationException;
+import org.omegat.util.Log;
+import org.omegat.util.NullBufferedWriter;
+import org.omegat.util.OStrings;
+import org.omegat.util.StringUtil;
+
+/**
+ * Filter for support Moodle PHP files.
+ * 
+ * Code adapted from the file: MozillaDTDFilter.java
+ *
+ * @author Enrique Estevez (keko.gl@gmail.com)
+ */
+public class MoodlePHPFilter extends AbstractFilter {
+
+    public static final String OPTION_REMOVE_STRINGS_UNTRANSLATED = "unremoveStringsUntranslated";
+
+    protected static final Pattern RE_ENTITY = Pattern.compile("\\$string\\['(.+)'] (=) '(.+)';",
+            Pattern.DOTALL);
+
+    protected Map<String, String> align;
+
+    /**
+     * If true, will remove non-translated segments in the target files
+     */
+    public static boolean removeStringsUntranslated = false;
+
+    @Override
+    public Instance[] getDefaultInstances() {
+        return new Instance[] { new Instance("*.php") };
+    }
+
+    @Override
+    public String getFileFormatName() {
+        return OStrings.getString("MOODLEPHP_FILTER_NAME");
+    }
+
+    @Override
+    public boolean isSourceEncodingVariable() {
+        return false;
+    }
+
+    @Override
+    public boolean isTargetEncodingVariable() {
+        return false;
+    }
+
+    @Override
+    protected BufferedReader createReader(File inFile, String inEncoding)
+            throws UnsupportedEncodingException, IOException {
+        return new BufferedReader(new InputStreamReader(new FileInputStream(inFile), StandardCharsets.UTF_8));
+    }
+
+    @Override
+    protected BufferedWriter createWriter(File outFile, String outEncoding)
+            throws UnsupportedEncodingException, IOException {
+        return new BufferedWriter(new OutputStreamWriter(new FileOutputStream(outFile), StandardCharsets.UTF_8));
+    }
+
+    @Override
+    protected void processFile(BufferedReader inFile, BufferedWriter outFile, FilterContext fc) throws IOException,
+            TranslationException {
+
+        String removeStringsUntranslatedStr = processOptions.get(OPTION_REMOVE_STRINGS_UNTRANSLATED);
+        // If the value is null the default is false
+        if ((removeStringsUntranslatedStr != null) && (removeStringsUntranslatedStr.equalsIgnoreCase("true"))) {
+            removeStringsUntranslated = true;
+        } else {
+            removeStringsUntranslated = false;
+        }
+
+        StringBuilder block = new StringBuilder();
+        boolean isInBlock = false;
+        boolean foundQuotes = false;
```

Comment:
`foundQuotes` isn't actually used for anything. If that's correct, please remove it.

Reply:
Remove it.